java/arcs/core/storage/StorageProxy.kt

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage
+
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtException
+import arcs.core.crdt.CrdtModel
+import arcs.core.crdt.CrdtOperation
+import arcs.core.crdt.internal.VersionMap
+import arcs.core.util.guardWith
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/** ValueAndVersion is a tuple of a value of some type and a [VersionMap] provided from its [StorageProxy]. */
```

Comment:
Square brackets around types: `[ValueAndVersion]`

But even better is to omit it entirely, since it's obvious what you're talking about: `A tuple of ...`

Reply:
Made the comment better:
/** [ValueAndVersion] is used to tie a value to the VersionMap it was current as-of. */